content/learn/beginner/inputs-1.md

```diff
@@ -0,0 +1,169 @@
++++
+title = "Inputs 1"
+weight = 240
++++
+
+When we were starting to write Custom Commands, we stuck to a somewhat static custom command --- one that doesn't take
+any arguments and has a fixed output. This is a good starting point, but it's not very useful in practice. In this
+chapter, we'll learn how to create custom commands that take inputs.
+
+## Parsing Arguments
+
+Because parsing arguments is no easy task, we provide a convenience function, `parseArgs`, to parse the arguments passed
+to a custom command. For simple commands, this is the recommended way. In a later chapter, we will explore a more
+hands-on approach to parsing arguments.
+
+### Defining Arguments
+
+The first step is to define the arguments that the command will take. This is done using the aforementioned `parseArgs`
+function. The syntax is as follows:
+
+```go
+{{ $args := parseArgs <"num required args"> <"error message"> ...carg }}
```

Comment:
The `<"...">` scheme is quite confusing. How about just `parseArgs required_args err_msg ...cargs`?

Reply:
Initially I was going for a `<required> [optional]` format, but
evidently I kind of lost track of doing that, as evident by the `...carg`
notation. Your proposed format seems reasonable.